File: Exercises2024/Chapter04_basic_mc/C/3-HardDisks/static/Source/writepdb.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>
#include "system.h"

// make a movie file of the simulation box
// use vmd to view it..
void WritePdb(FILE *FilePtr)
{
  int i;
  static int CountModel=0;

  CountModel++;
  fprintf(FilePtr,"%s %9d\n","MODEL",CountModel);

  fprintf(FilePtr,"CRYST1   %6.3f   %6.3f   %6.3f  90.00  90.00  90.00 P 1         1\n", 2.0*BOXSIZE, 2.0*BOXSIZE, 2.0*BOXSIZE);
  for(i=0;i<NumberOfParticles;i++)
  {
    fprintf(FilePtr,"%s%7d%s%12d    %8.3lf%8.3lf%8.3lf                      %2s\n",
      "ATOM",i,"  H",i,Positions[i].x*2.0,Positions[i].y*2.0,0.0," H");
  }
  fprintf(FilePtr,"%s\n","ENDMDL");
}

File: Exercises2024/Chapter04_basic_mc/Fortran/3-HardDisks/static/Source/writepdb.f
      Subroutine Writepdb
      Implicit None

      Include 'system.inc'
      
      Integer I,Countmodel
      Double Precision Z

      Data Countmodel/ 0/

      Save Countmodel

      Z          = 0.0d0
      Countmodel = Countmodel + 1

      Write(22,'(A,I9)') 'MODEL',Countmodel
      Write(22,'(A6, 3f9.3, 3f7.2, 1x, A)') 'CRYST1',
     &   2.0d0 * BoxSize, 2.0d0 * BoxSize, 2.0d0 * BoxSize,
     &   90.0, 90.0, 90.00, 'P 1         1'

      Do I=1,Npart

         Write(22,'(A6,I5,1x,A4,1x,3x,1x,4x,1x,4x,3f8.3,12x,10x,A2)')
     &        'ATOM  ',I,' H  ',
     &        2.0d0 * X(I),2.0d0 * Y(I), 2.0d0 * Z,' H'
      Enddo

      Write(22,'(A)') 'ENDMDL'

      Return
      End
